irqbalance.c

```diff
@@ -344,11 +352,16 @@ void get_object_stat(struct topo_obj *object, void *data)
 	if (!*stats) {
 		*stats = calloc(irqdlen + 31 + 11 + 20 + 11 + 1, 1);
 	} else {
-		*stats = realloc(*stats, strlen(*stats) + irqdlen + 31 + 11 + 20 + 11 + 1);
+		old = *stats;
+		*stats = realloc(old, strlen(old) + irqdlen + 31 + 11 + 20 + 11 + 1);
 	}
 
-	if (!*stats)
+	if (!*stats) {
+		/* if realloc fail, not free the original memory */
+		*stats = old;
+		free(irq_data);
 		return;
+	}
```

Comment:
same as the above comment, use a temp variable to capture the return of calloc/realloc, test it for null, and return if it is, or assign it to *stats otherwise

Reply:
ok, I have modify the patch. thanks.